apps/dotcms-ui-e2e/src/pages/DotSiteBrowser/DotSiteBrowser.ts

```diff
@@ -0,0 +1,66 @@
+import Form from '../../support/shared/Form';
+import Navigation from '../../support/shared/Navigation';
+import Page from '../../support/shared/Page';
+
+const URL = '/dotAdmin/#/c/site-browser';
+const CREATE_PAGE_BUTTON = '#addNewDropDownButtonDiv span.dijitButtonNode';
+const IFRAME = 'iframe#detailFrame';
+
+class DotSiteBrowser {
+    static openPage() {
+        cy.intercept('POST', 'BrowserAjax.openFolderContent.dwr').as('getLoggedinUser');
+        Navigation.visit(URL);
+        Navigation.assertPageUrlIs(URL);
+        cy.wait('@getLoggedinUser');
+    }
+
+    static checkSiteBrowserPageLoaded() {
+        Page.assertElementContains('.p-breadcrumb > ul > :nth-child(3)', `Browser`); // Header
+        cy.get(IFRAME).iframe(() => {
+            Page.assertElementSize(cy.get('#borderContainer'), 1); // File tree container
+            Page.assertElementSize(cy.get(CREATE_PAGE_BUTTON), 1); // Add action button
+        });
+    }
+
+    static openCreatePageDialog({ type }: { type: string }) {
+        cy.get(IFRAME).iframe().as('iframeContent');
+        cy.get('@iframeContent').then(() => {
+            Page.click(cy.get('@iframeContent').find(CREATE_PAGE_BUTTON));
```

Comment:
Shouldn't we save `cy.get('@iframeContent')` in a reusable var?

Reply:
prob, the thing with how iframes objects are grabed and then handle needs to be tested, cuz sometimes (based on the work I did) some unexpected behaviours happens!